core/services/gateway/connector/connector.go

```diff
@@ -125,6 +126,22 @@ func NewGatewayConnector(config *ConnectorConfig, signer Signer, handler Gateway
 	return connector, nil
 }
 
+func (c *gatewayConnector) AddHandler(methods []string, handler GatewayConnectorHandler) error {
+	if handler == nil {
+		return errors.New("cannot add a nil handler")
+	}
+	for _, method := range methods {
+		if _, exists := c.handlers[method]; exists {
```

Comment:
why aren't you getting index from the range and then using that on c.handlers?

Reply:
c.handlers map is indexed by strings, not integers. Also here I only check for existence before adding later in the subsequent loop.